### ISSUE TITLE
coral edge tpu performance

### ISSUE THREAD

**Issue by Pepslee** [ultralytics/ultralytics#18615]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I`m using raspberry pi 5 (compute module on dev board) and coral. 
Step to convert model in docker ultralytics/ultralytics:latest:
```shell
yolo export model=yolo11n.pt format=edgetpu imgsz=320 
```

At the output I have file `yolo11n_full_integer_quant_edgetpu.tflite`

Run the model on raspberry:
```python
from ultralytics import YOLO

model = YOLO("yolo11n_full_integer_quant_edgetpu.tflite") 
model.predict("street.jpg", imgsz=320)
```
Output:
`WARNING ⚠️ Unable to automatically guess model task, assuming 'task=detect'. Explicitly define task for your model, i.e. 'task=detect', 'segment', 'classify','pose' or 'obb'.
Loading yolo11n_320_full_integer_quant_edgetpu.tflite on device 0 for TensorFlow Lite Edge TPU inference...

image 1/1 /home/pi/coral_test/street.jpg: 320x320 5 persons, 2 cars, 1 truck, 53.4ms
Speed: 2.5ms preprocess, 53.4ms inference, 1.8ms postprocess per image at shape (1, 3, 320, 320)`

So inference time is 53 ms.

There is benchmark [table](https://docs.ultralytics.com/guides/coral-edge-tpu-on-raspberry-pi/#benchmarks) in the doc where I can see that 320 image size yolo11n inference time is 22 ms.

Coral is connected over usb3.0:
```shell
lsusb
Bus 005 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 004 Device 005: ID 18d1:9302 Google Inc.
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```
Can someone give me a hint, what I`m doing wrong ? and how to improve performance ?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Pepslee, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users, where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it further. Based on the information you’ve shared, this could include details such as exact hardware specifications, how the Coral Edge TPU is connected, and any additional logs or outputs you might observe during inference. 

For benchmarking or performance ❓ Questions, please ensure you’ve followed the [Coral Edge TPU Guide](https://docs.ultralytics.com/guides/coral-edge-tpu-on-raspberry-pi/) closely, particularly around USB 3.0 connectivity and image optimization. It might also help to confirm whether other benchmarked models run as expected on your hardware for comparison.

To improve performance, try the following:
1. Upgrade to the latest version of the `ultralytics` package and dependencies. You can do so using:
```bash
pip install -U ultralytics
```
2. Confirm whether your Coral Edge TPU and Raspberry Pi 5 hardware configurations match those used in our [benchmark table](https://docs.ultralytics.com/guides/coral-edge-tpu-on-raspberry-pi/#benchmarks).
3. Test using a smaller input image size (`imgsz`) or experiment with other image preprocessing techniques to optimize speed.

You’re not alone! Join the community that best suits your engagement style:
- For real-time support or discussions, hop onto our [Discord](https://discord.com/invite/ultralytics) server 🎧.
- Prefer forum-style threads? Use [Discourse](https://community.ultralytics.com).
- Or share and explore insights in our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Use the latest `ultralytics` framework with all its [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch >=1.8**](https://pytorch.org/get-started/locally/) to ensure your issue isn’t already fixed in the latest release.

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be executed in verified, up-to-date environments (complete with all dependencies):  
- **Notebooks** with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** VMs: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon AWS**: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: Refer to the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests ensure functionality across all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) 24/7 on macOS, Windows, and Ubuntu.

Please note that this is an automated response 🤖, and an Ultralytics engineer will also be with you soon to assist further 🚀.

**Comment by Pepslee** [comment]
@Y-T-G Thanks, I have taken this into account. I run 50 + iterations but inference time still around 50+ ms.  That one run in example is just for short and easy to read example.

**Comment by Skillnoob** [comment]
The table is created with yolov8, not yolo11, it says that in the table as well. I've seen some other users also report v11 being slow, but I need to set up a new OS for my pi first before I can test it.

**Comment by Pepslee** [comment]
@Skillnoob Yes, I just checked Yolov8n, and it is around 20 ms inference time. Thank you for the hint.

**Comment by Vincenzo220806** [comment]
hi I followed this video: https://youtu.be/lJRLzXf7EVs?si=NnPgkaUit1Bavr-4 to use the coral usb acceleretor in real time on my raspberry pi5 but I can't use the cam with the flat by downloading the picamera2 library in my virtual environment thonny tells me that it can't find the libcamera package but as soon as I write "libcamera-hello" on the virtual environment terminal I can easily use the cam I also tried to give access to the system packages with: "--system-site-packages" but I still can't solve the problem could someone help me

**Comment by glenn-jocher** [comment]
For picamera2 issues in Thonny's virtual environment, ensure system packages are installed first:  
```bash
sudo apt install python3-libcamera python3-picamera2
```  
Then recreate your virtual environment with `--system-site-packages`. This resolves missing dependencies while maintaining venv isolation. Let us know if you still encounter issues.
